workflows/run.nf

```diff
@@ -30,6 +30,16 @@ workflow RUN {
     kraken_db_path = "${params.ref_dir}/results/kraken_db"
     blast_db_path = "${params.ref_dir}/results/${params.blast_db_prefix}"
 
+    // Check if grouping column exists in samplesheet
+    check_grouping = new File(params.sample_sheet).text.readLines()[0].contains('group') ? true : false
+    if (params.grouping != check_grouping) {
+        if (params.grouping && !check_grouping) {
+            throw new Exception("Grouping column not found in samplesheet")
+        } else if (!params.grouping && check_grouping) {
+            throw new Exception("Grouping column found in samplesheet but not requested")
```

Comment:
What do you think about making this a warning? In principle people might genuinely want to run an ungrouped analysis off a grouped samplesheet.

Reply:
I think it's better to have an exception as, in my experience, people tend to ignore warnings or just may miss them. The user can make another samplesheet without the grouping column if they wish.